### PR TITLE
Add Math sqrt overloads for Bigs

### PR DESCRIPTION
**Pull request by iSarCasm**
This PR references #4792 

It adds `Math#sqrt` overloads when we `require "big"`

### CODE REVIEW

File: src/big/math.cr [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require "big"
+
+module Math
```
Comment (RX14) [inline src/big/math.cr]: Please move this into `Math` overloads at the end of each of the respective `big_*.cr` files like [here](https://github.com/crystal-lang/crystal/blob/f0ec21e1209ba92ce67f4900a0a679cd3f92d0cd/src/big/big_float.cr#L280).

File: spec/std/math_spec.cr [hunk 2]
```diff
@@ -41,6 +42,18 @@ describe "Math" do
       Math.sqrt(4_f32).should eq(2)
       Math.sqrt(4).should eq(2)
     end
+
```
Comment (straight-shoota) [inline spec/std/math_spec.cr]: Maybe this should better be in the big* spec?

Comment (iSarCasm) [reply spec/std/math_spec.cr]: @straight-shoota Idk, but spec for json+big and yaml+big are under json and yaml

File: spec/std/math_spec.cr [hunk 3]
```diff
@@ -1,4 +1,5 @@
 require "spec"
+require "big"
```
Comment (RX14) [inline spec/std/math_spec.cr]: You can remove this line now.

File: spec/std/math_spec.cr (hunk 2 above)
Comment (RX14) [reply spec/std/math_spec.cr]: I think this is better in the `big` specs.

File: spec/std/math_spec.cr (hunk 3 above)
Comment (iSarCasm) [reply spec/std/math_spec.cr]: @RX14 I get `Error in spec/std/math_spec.cr:46: undefined constant BigFloat` then

**Comment by asterite** [comment]
All of the `Big*` types are kind of related, so it doesn't matter where things are defined, all `Big*` types should be `required`d in one go with `require "big"`. We should remove/hide those `big/big_float`.

**Comment by iSarCasm** [comment]
@asterite not quite sure I got what you meant. @RX14 does it look better now?

**Comment by RX14** [comment]
@asterite Currently, each `BigFoo` type can be required individually. If you want to clean up the code and make it so that isn't possible go ahead but thats outside the scope of this PR.

**Pull request merged**